File: leetcode/0819-most-common-word/mostCommonWord.test.ts
import cases from "./cases.json";
import mostCommonWord from "./mostCommonWord";

test("most common word", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = mostCommonWord(
      testCase.input.paragraph,
      testCase.input.banned
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toBe(testCase.output);
  });
});

File: leetcode/0819-most-common-word/mostCommonWord_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type MostCommonWordCase struct {
	Input struct {
		Paragraph string
		Banned    []string
	}

	Output string
}

func TestMostCommonWord(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MostCommonWordCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := mostCommonWord(testCase.Input.Paragraph, testCase.Input.Banned)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
